packages/react-core/src/components/Nav/NavItem.tsx

```diff
@@ -34,6 +34,8 @@ export interface NavItemProps extends Omit<React.HTMLProps<HTMLAnchorElement>, '
   onShowFlyout?: () => void;
   /** @beta Opt-in for updated popper that does not use findDOMNode. */
   removeFindDomNode?: boolean;
+  /** z-index of the nav item */
+  zIndex?: number;
```

Comment:
Tweak to the above comment: looks like this would only apply if the `flyout` prop is also passed.